packages/shared/src/components/comments/CommentInputOrPage.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { ReactElement } from 'react';
+import { useRouter } from 'next/router';
+import {
+  CommentMarkdownInput,
+  CommentMarkdownInputProps,
+} from '../fields/MarkdownInput/CommentMarkdownInput';
+import { ViewSize, useViewSize } from '../../hooks';
+
+interface CommentInputOrPageProps
+  extends Omit<CommentMarkdownInputProps, 'className'> {
+  onClose?: () => void;
+  className?: {
+    input?: CommentMarkdownInputProps['className'];
+    modal?: string;
+  };
+  replyToCommentId?: string;
+}
+
+export default function CommentInputOrPage({
+  onClose,
+  className,
+  ...props
+}: CommentInputOrPageProps): ReactElement {
+  const isMobile = useViewSize(ViewSize.MobileL);
+  const router = useRouter();
+
+  if (isMobile) {
```

Comment:
Would it be better to actually keep this component clean and only render this component if !isMobile ? we can do this where we call the component

Reply:
I see though we call the component in many places so we'd have to replicate logic if we do it like that

Reply:
Yeah, I wanted to avoid adding the isMobile check to all the callers. We had a similar issue before with the modals as well and decided to use a wrapper to decide what to render based on the view width.